### ISSUE TITLE
Transfer host protocol

### ISSUE THREAD

**Issue by AdmiralGhoster** [Frooxius/NeosPublic#258]
If its possible it would be great if you could have a way of transferring who is hosting the world while the world is active
The restriction would be that the next world host must have admin permissions and be active in the world to allow the transfer. Also a pop up should appear on both the current and next host agreeing to the transfer.

This way if a user has a not to great connection or pc, they can pass off the hosting to another person. 

Part of this would also be a check on the current hosts part to make sure all clients have successfully moved hosts and continue to redirect clients to the new host untill there is only that person left then move themselfs while updating the world sphere to reflect the new host.

The last part should  be a user popup that appears telling everyone individually that a host transfer is in progress and when they have switched and finally once the whole transfer is complete.

**Comment by sirkitree** [comment]
Agreed. This would also be useful in cases where the host has to leave and the group wants to continue without dropping the world.

**Comment by Medravr** [comment]
Isn't this an already planned feature? Either way, I cast my vote. :) Many would benefit greatly as quite frequently hosts need to reset or leave for various reasons.

**Comment by Frooxius** [comment]
This is planned at some point, but probably won't appear anytime soon since there's a fair bit of work involved and you can just manually go to a different world. Being able to host on a dedicated server is a higher priority.

There's multiple approaches to implement this too. A more lazy, but less efficient one is to effectively pause everyone, save the world, load another instance, connect everyone and switch. A lot more involved one would essentially keep the same session, but that's a lot more involved.

**Comment by AdmiralGhoster** [comment]
The reason i made this was for the other feature i would like to see aka #259
